.github/workflows/run_tests.yaml

```diff
@@ -0,0 +1,130 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+# Official DOc on GitHub Workflow Syntax: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+
+# Conda based env testing: https://github.com/conda-incubator/setup-miniconda
+name: Run tests
+
+on:
+
+  # Trigger the workflow on push but only for the master branch
+  push:
+    branches:
+      - master
+
+  # Trigger current workflow on PR from ANY branch
+  pull_request:
+    branches:
+      - '*'
+
+  # Cron Trigger
+  schedule:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onschedule
+    # i.e. this runs every day at 4:33AM for the master branch.
+    - cron: '33 4 * * *'
+
+env:
+  # Even when given -y, apt will still sometimes hang at a prompt if a package
+  # has clarifications to ask; DEBIAN_FRONTEND=noninteractive prevents that,
+  # This will be defined for non-debian platforms below too, but there's no harm in that.
+  # (TravisCI quietly defined this on all their platforms, but we have to give it manually on GithubCI.)
+  DEBIAN_FRONTEND: 'noninteractive'
+
+jobs:
+  ultra_matrix_test:
+    name: Matrix Test of Python ${{ matrix.python-version }} on ${{ matrix.os }}
+
+    # Matrix driven OS
+    runs-on: ${{ matrix.os }}
+
+    # Default shell for ALL subsequent steps.
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    # Defining matrix for OS and Python
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ "ubuntu-latest", "macos-latest", "windows-latest" ]
+        python-version: [ "3.7", "3.8", "3.9" ]
+
+    # Main steps for the test to be reproduced across OS x Python
+    steps:
+      # Step 0: Checkout code.
+      - uses: actions/checkout@v2
+
+      # Step 1: List main repo by time modified, and print path
+      - name: List all current repository files and modification date
+        # Parameters.
+        run: |
+          ls -lasth
+          pwd
+
+      # Step 2: Install environment.
+      - name: Install Conda Env All OS
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: ${{matrix.python-version}}
+          activate-environment: ads_venv
+          auto-update-conda: true
+          environment-file: environment.yml
```

Comment:
Discussing in meeting right now about this, we would like to test with each OS-specific environment file separately.
In that case, it would probably only works with Python 3.7.

Reply:
Updating PR shortly. 

> 1 - set default again in the environment.yml (we can discuss this in another issue)

👍 being done. 

> 2 - Restrict to Python 3.7 again (more testing would be necessary and we'd need to have a more stable/less restrictibe environment file I think)

👍 being done.  3.7 only for both setup.py and test

> 3 - (I think) Change the build badge in the README, as it currently points to Travis and we should have it for the GitHub actions I think.

Thanks for calling this out. I just updated in the readme. Turned out it was very straightforward per this doc: https://docs.github.com/en/actions/managing-workflow-runs/adding-a-workflow-status-badge 

> 4 - Last thing I think needs to be done is outside of the code, in the repo's settings. I'm assuming I'll have to revoke Travis-CI access/PR checks in the settings, but is there anything I need to add for the GitHub actions to run as conditions for PRs?

Yeah, been a while. I am not toto sure. We MIGHT have to also remove the .travis.yml as well as revoke access? Never done this part before.  Maybe someone who has a bit more experience @joshuacwnewton ? Did you guys have to do yml removal first or revoke access?